ext/hash/hash.stub.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/** @return string|false */
+function hash(string $algo, string $data, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_file(string $algo, string $filename, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_hmac(string $algo, string $data, string $key, bool $raw_output = false) {}
+
+/** @return string|false */
+function hash_hmac_file(string $algo, string $data, string $key, bool $raw_output = false) {}
+
+function hash_init(string $algo, int $options = 0, ?string $key = null): HashContext {}
```

Comment:
The `$key` must not be `null` (C NULL != PHP NULL), so
```suggestion
function hash_init(string $algo, int $options = 0, string $key = UNKNOWN): HashContext {}
```